video/vdu_stream_processor.h

```diff
@@ -155,9 +163,9 @@ int16_t VDUStreamProcessor::readByte_t(uint16_t timeout = COMMS_TIMEOUT) {
 //
 int32_t VDUStreamProcessor::readWord_t(uint16_t timeout = COMMS_TIMEOUT) {
 	auto l = readByte_t(timeout);
-	if (l >= 0) {
+	if (l != -1) {
```

Comment:
these comparisons, checking for `!= -1` rather than `>=0` provided a partial fix to the `-O3` compiler flag issues we had been seeing.

as there is no harm in keeping these changes, they have remained